app/controllers/nodes_controller.rb

```diff
@@ -18,6 +18,42 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  # rubocop:disable Metrics/AbcSize
+  def assign
+    @minion = Minion.find(params[:node_id])
+
+    unless Minion.roles.keys.include?(params[:role])
+      raise Minion::InvalidRole, "Could not assign role #{params[:role]}"
+    end
+
+    respond_to do |format|
+      if @minion.assign_role(params[:role])
+        res, body = Velum::Salt.orchestrate
+        http_status = res.code == "200" ? :ok : :unprocessable_entity
+        flash[:error] = "Salt failed to orchestrate: #{body.inspect}" unless http_status == :ok
+
+        format.html { redirect_to nodes_path }
+        format.json { head http_status }
+      else
+        format.html do
+          flash[:error] = @minion.errors.full_messages.first
+          redirect_to nodes_path
+        end
+        format.json { render json: @minion.errors, status: :unprocessable_entity }
+      end
+    end
+  rescue Velum::SaltApi::SaltConnectionException,
+         ActiveRecord::RecordNotFound,
```

Comment:
This rescues a quite bit block of code. Inside that block there might be more than one places where this exception can occur and some of them we might not even know right now. If we want a beautiful error in case the user tampers with the params maybe it would be better to explicitly check for that case at line 23 and let the rest of the code raise Exceptions.